controllers/mover/syncthing/mover.go

```diff
@@ -0,0 +1,751 @@
+					{
+						Port:       syncthingDataPort,
+						TargetPort: intstr.FromInt(syncthingDataPort),
+						Protocol:   "TCP",
+						Name:       "syncthing-data",
+					},
+				},
+				Type: m.serviceType,
+			},
+		}
+
+		if err := ctrl.SetControllerReference(m.owner, service, m.client.Scheme()); err != nil {
+			m.logger.V(3).Error(err, "failed to set owner reference")
+			return nil, err
+		}
+		if err := m.client.Create(ctx, service); err != nil {
+			m.logger.Error(err, "error creating the service")
+			return nil, err
+		}
+	}
+	return service, nil
+}
+
+// GetDataServiceAddress Will return a string representing the address of the data service, prefixed with TCP.
+func (m *Mover) GetDataServiceAddress(service *corev1.Service) (string, error) {
+	// format the address based on the type of service we're using
+	// supported service types: ClusterIP, LoadBalancer
+	address := utils.GetServiceAddress(service)
+	if address == "" {
+		return "", fmt.Errorf("could not get an address for the service")
+	}
+	address = "tcp://" + address + ":" + strconv.Itoa(syncthingDataPort)
+	return address, nil
+}
+
+// Cleanup will remove any resources that were created by the mover.
+// This is currently a no-op since Syncthing is always-on.
+func (m *Mover) Cleanup(ctx context.Context) (mover.Result, error) {
+	err := utils.CleanupObjects(ctx, m.client, m.logger, m.owner, []client.Object{})
+	if err != nil {
+		return mover.InProgress(), err
+	}
+	return mover.Complete(), nil
+}
+
+// configureSyncthingAPIClient Configures the Syncthing API client if it has not been configured yet.
+func (m *Mover) configureSyncthingAPIClient(apiSecret *corev1.Secret) error {
+	// set the api key
+	m.syncthing.APIConfig.APIKey = string(apiSecret.Data["apikey"])
+
+	// load the TLS certificate
+	clientConfig, err := m.loadTLSConfigFromSecret(apiSecret)
+	if err != nil {
+		return err
+	}
+	m.syncthing.APIConfig.TLSConfig = clientConfig
+
+	// create a new client or use the existing one
+	m.syncthing.APIConfig.Client = m.syncthing.APIConfig.BuildOrUseExistingTLSClient()
+
+	return nil
+}
+
+// ensureIsConfigured Makes sure that the Syncthing config is up-to-date.
+func (m *Mover) ensureIsConfigured(apiSecret *corev1.Secret) error {
+	// This function should make sure to ignore all of devices (leave them as they are) introduced to us by other nodes.
+	// This is important because Syncthing will otherwise reconfigure them on its own, creating a cycle where
+	// VolSync removes a device, only to be re-added by Syncthing, only to be removed again by VolSync, etc.
+
+	// reconciles the Syncthing object
+	err := m.syncthing.FetchLatestInfo()
+	if err != nil {
+		return err
+	}
+	m.logger.V(4).Info("Syncthing config", "config", m.syncthing.Config)
+
+	// make sure that the spec doesn't try to add a previously introduced node
+	if m.syncthing.PeerListContainsIntroduced(m.peerList) {
+		return fmt.Errorf("the peer list contains a node that has been introduced to us by another node")
```

Comment:
We don't want users to be explicitly connecting nodes that have already been introduced, since the introducers are treated as hubs & non-introducers are spokes. By erroring out here, we are asserting that we cannot directly connect with other devices being introduced to us, so that they can be disconnected from the cluster by having their listing removed in the introducer node's `peerList`. 

See: https://docs.syncthing.net/users/introducer.html